src/Files.App/Views/Pages/Properties.xaml.cs

```diff
@@ -1,269 +1,282 @@
-            }
-        }
-    }
+	public sealed partial class Properties : Page
+	{
+		private CancellationTokenSource? tokenSource = new CancellationTokenSource();
+		private ContentDialog propertiesDialog;
+
+		private object navParameterItem;
+		private IShellPage AppInstance;
+
+		private ListedItem listedItem;
+
+		public SettingsViewModel AppSettings => App.AppSettings;
+
+		public AppWindow appWindow;
+
+		public Properties()
+		{
+			InitializeComponent();
+
+			var flowDirectionSetting = /*
+				TODO ResourceContext.GetForCurrentView and ResourceContext.GetForViewIndependentUse do not exist in Windows App SDK
+				Use your ResourceManager instance to create a ResourceContext as below. If you already have a ResourceManager instance,
+				replace the new instance created below with correct instance.
+				Read: https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/migrate-to-windows-app-sdk/guides/mrtcore
+			*/new Microsoft.Windows.ApplicationModel.Resources.ResourceManager().CreateResourceContext().QualifierValues["LayoutDirection"];
+
+			if (flowDirectionSetting == "RTL")
+				FlowDirection = FlowDirection.RightToLeft;
+		}
+
+		protected override void OnNavigatedTo(NavigationEventArgs e)
+		{
+			var args = e.Parameter as PropertiesPageNavigationArguments;
+			AppInstance = args.AppInstanceArgument;
+			navParameterItem = args.Item;
+			listedItem = args.Item as ListedItem;
+			TabShorcut.Visibility = listedItem != null && listedItem.IsShortcutItem ? Visibility.Visible : Visibility.Collapsed;
+			TabLibrary.Visibility = listedItem != null && listedItem.IsLibraryItem ? Visibility.Visible : Visibility.Collapsed;
+			TabDetails.Visibility = listedItem != null && listedItem.FileExtension != null && !listedItem.IsShortcutItem && !listedItem.IsLibraryItem ? Visibility.Visible : Visibility.Collapsed;
+			TabSecurity.Visibility = args.Item is DriveItem ||
+				(listedItem != null && !listedItem.IsLibraryItem && !listedItem.IsRecycleBinItem) ? Visibility.Visible : Visibility.Collapsed;
+			TabCustomization.Visibility = listedItem != null && !listedItem.IsLibraryItem && (
+				(listedItem.PrimaryItemAttribute == Windows.Storage.StorageItemTypes.Folder && !listedItem.IsZipItem) ||
+				(listedItem.IsShortcutItem && !listedItem.IsLinkItem)) ? Visibility.Visible : Visibility.Collapsed;
+			TabCompatibility.Visibility = listedItem != null && (
+					".exe".Equals(listedItem is ShortcutItem sht ? System.IO.Path.GetExtension(sht.TargetPath) : listedItem.FileExtension, StringComparison.OrdinalIgnoreCase)
+				) ? Visibility.Visible : Visibility.Collapsed;
+			base.OnNavigatedTo(e);
+		}
+
+		private async void Properties_Loaded(object sender, RoutedEventArgs e)
+		{
+			AppSettings.ThemeModeChanged += AppSettings_ThemeModeChanged;
+			if (ApiInformation.IsApiContractPresent("Windows.Foundation.UniversalApiContract", 8))
+			{
+				SetDragRegion();
+				appWindow.Destroying += AppWindow_Destroying;
+				appWindow.Changed += AppWindow_Changed;
```

Comment:
Shouldn't this be the size changed event?

Reply:
If you meant subscribing the new method to `SizeChanged` event, I can't do that as AppWindow events are only `Closing`, `Destroying` and `Changed`.
I would not rename `AppWindow_Changed` as this is a general event, raised in many cases. 
Because of the above, I'll rename `SetDragRegion` to `AppWindow_SizeChanged`

Reply:
That's correct, we do something similar on the MainPage